src/network/networkd-route-nexthop.c

```diff
@@ -115,12 +116,12 @@ static int route_nexthop_compare_func_full(const RouteNextHop *a, const RouteNex
         if (r != 0)
                 return r;
 
-        if (!IN_SET(a->family, AF_INET, AF_INET6))
-                return 0;
-
-        r = memcmp(&a->gw, &b->gw, FAMILY_ADDRESS_SIZE(a->family));
-        if (r != 0)
-                return r;
+        /* We don't have the gateway yet for device-only routes, which are AF_UNSPEC */
```

Comment:
Please do not workaround here. But adjust family in route_section_verify_nexthops().